src/BuiltInTools/dotnet-watch/HotReload/NamedPipeContract.cs

```diff
@@ -79,117 +63,369 @@ public static void WriteLog(Stream stream, IReadOnlyCollection<(string message,
         /// </summary>
         public static async ValueTask<UpdatePayload> ReadAsync(Stream stream, CancellationToken cancellationToken)
         {
-            using var binaryReader = new BinaryReader(stream, Encoding.UTF8, leaveOpen: true);
-            var version = binaryReader.ReadByte();
+            var version = await stream.ReadByteAsync(cancellationToken);
             if (version != Version)
             {
                 throw new NotSupportedException($"Unsupported version {version}.");
             }
 
-            var count = binaryReader.ReadInt32();
+            var count = await stream.ReadInt32Async(cancellationToken);
 
             var deltas = new UpdateDelta[count];
             for (var i = 0; i < count; i++)
             {
-                var moduleId = Guid.Parse(binaryReader.ReadString());
-                var metadataDelta = await ReadBytesAsync(binaryReader, cancellationToken);
-                var ilDelta = await ReadBytesAsync(binaryReader, cancellationToken);
-                var pdbDelta = await ReadBytesAsync(binaryReader, cancellationToken);
-                var updatedTypes = ReadIntArray(binaryReader);
+                var moduleId = await stream.ReadGuidAsync(cancellationToken);
```

Comment:
Is this an area where checking the CancellationToken is needed?

Reply:
Not sure why would it be needed. We are calling to Stream.ReadAsync, which will use the cancellation token while waiting for the data.